File: app/src/main/java/net/squanchy/tweets/domain/view/TweetViewModel.java
package net.squanchy.tweets.domain.view;

import com.google.auto.value.AutoValue;

import java.util.List;

@AutoValue
public abstract class TweetViewModel {

    public abstract long id();

    public abstract String text();

    public abstract CharSequence spannedText();

    public abstract User user();

    public abstract String createdAt();

    public abstract List<String> mediaUrls();

    public static Builder builder() {
        return new AutoValue_TweetViewModel.Builder();
    }

    @AutoValue.Builder
    public abstract static class Builder {

        public abstract Builder id(long id);

        public abstract Builder text(String text);

        public abstract Builder spannedText(CharSequence spannedText);

        public abstract Builder user(User user);

        public abstract Builder createdAt(String createdAt);

        public abstract Builder mediaUrls(List<String> mediaUrls);

        public abstract TweetViewModel build();
    }
}

File: app/src/main/java/net/squanchy/tweets/domain/view/TweetViewModel.kt
package net.squanchy.tweets.domain.view

import com.google.firebase.Timestamp
import net.squanchy.tweets.domain.TweetLinkInfo

data class TweetViewModel(
    val id: Long,
    val text: String,
    val spannedText: CharSequence,
    val user: User,
    val createdAt: Timestamp,
    val photoUrl: String?,
    val linkInfo: TweetLinkInfo
)
